### PR TITLE
Extend the Functionality of `Faculty`

### LINKED ISSUE #87: Update `Faculty` class values to allow for Synonyms
example: SoC, School of Computing, Computing should all be accpected user input

### PR DESCRIPTION
**Pull request by JerryWang0000**
Now several different Strings can refer to the same `Faculty` object, while the other files are not affected. Still, the valid values include only real faculties in NUS.

Closes #87

### CODE REVIEW

File: src/main/java/staffconnect/model/person/Faculty.java [hunk 1]
```diff
@@ -74,17 +77,21 @@ public Faculty(String faculty) {
...
-                return faculty;
-            }
+        Optional<FacultyName> match = Stream.of(FacultyName.values())
+                .filter(member -> Arrays.stream(member.facultyNameValues)
+                        .anyMatch(value -> value.equalsIgnoreCase(name)))
+                .findFirst();
+        if (match.isPresent()) {
+            return match.get();
+        } else {
+            throw new IllegalArgumentException("No enum constant matches the provided name: " + name);
```
Comment (Pluiexo) [inline src/main/java/staffconnect/model/person/Faculty.java]: Isn't it better to simplify this to use the ```orElseThrow``` method of the optional class?

**Review by Pluiexo — COMMENTED** [review]
Lgtm! Though minor nits with your usage of optionals!

File: src/main/java/staffconnect/model/person/Faculty.java (hunk 1 above)
Comment (JerryWang0000) [reply src/main/java/staffconnect/model/person/Faculty.java]: Yeah that makes sense

**Review by Pluiexo — APPROVED** [review]
LGTM!

**Pull request merged**